### PR TITLE
Integrate alpine `noarch` packages into other architectures index

### LINKED ISSUE #26691: Alpine Package Registry: Handling of 'noarch' packages
### Description

Hello,

thank you for integrating the possibility to use gitea as an alpine package registry.

When adding a package with the architecture 'noarch' (like the [salt package](https://dl-cdn.alpinelinux.org/alpine/edge/community/x86_64/salt-3006.1-r0.apk)) it can not be installed via apk.

The reason is the architecture ist automatically detected from the package in the source-code and the uploaded package is then only available under the 'noarch' archtitecture and not under the other architectures like 'x86_64'.

A short test with the salt package from edge to my gitea instance:

```
root@alpine-dev ~/test # wget https://dl-cdn.alpinelinux.org/alpine/edge/community/x86_64/salt-3006.2-r0.apk
Connecting to dl-cdn.alpinelinux.org ([2a04:4e42:6f::644]:443)
saving to 'salt-3006.2-r0.apk'
salt-3006.2-r0.apk   100% |**************************************************************************************************************************| 5625k  0:00:00 ETA
'salt-3006.2-r0.apk' saved
root@alpine-dev ~/test # curl --user testuser:password --upload-file salt-3006.2-r0.apk https://[***REDACTED***]/api/packages/alpine-packages/alpine/edge/
root@alpine-dev ~/test # apk update
fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/community/x86_64/APKINDEX.tar.gz
fetch https://[***REDACTED***]/api/packages/alpine-packages/alpine/edge/salt/x86_64/APKINDEX.tar.gz
WARNING: updating and opening https://[***REDACTED***]/api/packages/alpine-packages/alpine/edge/salt: No such file or directory
v3.18.3-55-g2ee93b9273a [https://dl-cdn.alpinelinux.org/alpine/v3.18/main]
v3.18.3-63-g5d6ac0a2673 [https://dl-cdn.alpinelinux.org/alpine/v3.18/community]
2 unavailable, 0 stale; 20063 distinct packages available
root@alpine-dev ~/test # wget https://[***REDACTED***]/api/packages/alpine-packages/alpine/edge/salt/x86_64/APKINDEX.tar.gz
Connecting to [***REDACTED***] ([***REDACTED***]:443)
wget: server returned error: HTTP/1.1 404 Not Found
root@alpine-dev ~/test # wget https://[***REDACTED***]/api/packages/alpine-packages/alpine/edge/salt/noarch/APKINDEX.tar.gz
Connecting to [***REDACTED***] ([***REDACTED***]:443)
saving to 'APKINDEX.tar.gz'
APKINDEX.tar.gz      100% |**************************************************************************************************************************|  1170  0:00:00 ETA
'APKINDEX.tar.gz' saved
```

Possible solutions to this problem from my point of view are:
- add 'noarch' packages to all architectures
- allow to set the architecture on upload

With regards
Joniw

### Gitea Version

1.20.3

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Test instance in Kubernetes

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by KN4CK3R**
Fixes #26691
Revert #24972

The alpine package manager expects `noarch` packages in the index of other architectures too.

### CODE REVIEW

**Comment by Xinayder** [comment]
This is a nice way to fix the issue with noarch packages in the Alpine registry. I used a different approach in Forgejo, based on the information gathered on the official Alpine channels that noarch packages are literally just copied to the available architectures in the repository. 

However, one issue I had was when I uploaded a noarch package to a repository that doesn't exist. I believe the same issue is present in your PR, if the repo is new and the first package you're uploading is a noarch package, there are no available architectures on the repository and the noarch package won't be copied to the other architectures. 

The other thing I can say is, while it's nice that you keep track of the noarch packages on a separate "repo", it's not something done by Alpine, since there is no noarch repository on the official registry and apk doesn't support it yet.

**Comment by KN4CK3R** [comment]
> there are no available architectures on the repository and the noarch package won't be copied to the other architectures

Yes, I know about the problem if there is no other package which provides an architecture. I decided to ignore it because I don't want to set some default architecture which may be correct or incorrect, the system just can't know about it.

> and apk doesn't support it yet.

Yes, maybe one client does in the future.

File: services/packages/alpine/repository.go [hunk 1]
```diff
@@ -23,14 +23,18 @@ import (
...
 	packages_module "code.gitea.io/gitea/modules/packages"
 	alpine_module "code.gitea.io/gitea/modules/packages/alpine"
 	"code.gitea.io/gitea/modules/util"
 	packages_service "code.gitea.io/gitea/services/packages"
 )
 
-const IndexFilename = "APKINDEX.tar.gz"
+const (
+	IndexFilename       = "APKINDEX"
+	IndexArchivFilename = IndexFilename + ".tar.gz"
```
Comment (lafriks) [inline services/packages/alpine/repository.go]:
```suggestion
	IndexArchiveFilename = IndexFilename + ".tar.gz"
```

**Review by silverwind — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

**Review by lunny — CHANGES_REQUESTED** [review]
The typo @KN4CK3R mentioned should be fixed before merging.

**Review by lunny — APPROVED** [review]

**Pull request merged**